apps/frontend/src/app/5_pages/MarketMakingPage/components/BobWIthdrawModal/BobWithdrawModal.tsx

```diff
@@ -9,95 +10,188 @@ import {
-const BALANCE_B = Decimal.from(0.06);
-
 type BobWithdrawModalProps = {
   isOpen: boolean;
   onClose: () => void;
+  pool: AmbientLiquidityPool;
+  position: AmbientPosition;
 };
 
 export const BobWithdrawModal: FC<BobWithdrawModalProps> = ({
   isOpen,
   onClose,
+  pool,
+  position,
 }) => {
+  const { croc } = useCrocContext();
+
+  const [depositedAmountBase, setDepositedAmountBase] = useState(Decimal.ZERO);
+  const [depositedAmountQuote, setDepositedAmountQuote] = useState(
+    Decimal.ZERO,
+  );
   const [withdrawAmount, setWithdrawAmount] = useState(Decimal.ZERO);
   const [secondaryWithdrawAmount, setSecondaryWithdrawAmount] = useState(
     Decimal.ZERO,
   );
+  const [withdrawLiquidity, setWithdrawLiquidity] = useState(Decimal.ZERO);
+  const [poolPrice, setPoolPrice] = useState(0);
+  const { poolTokens } = useGetPoolInfo(pool.base, pool.quote);
+
+  const currentLiquidity = useMemo(
+    () =>
+      position.positionType === PoolPositionType.ambient
+        ? position.ambientLiq
+        : position.positionType === PoolPositionType.concentrated
+        ? position.concLiq
+        : 0,
+    [position],
+  );
+
+  const isFullWithdrawal = useMemo(
+    () => withdrawAmount.eq(depositedAmountBase),
+    [withdrawAmount, depositedAmountBase],
+  );
 
-  const submitHandler = useCallback(() => {
-    console.log(
-      `Withdrawing ${withdrawAmount} ${POOL.assetA} and ${secondaryWithdrawAmount} ${POOL.assetB}`,
-    );
-  }, [secondaryWithdrawAmount, withdrawAmount]);
+  const withdraw = useMemo(
+    () =>
+      isFullWithdrawal
+        ? BigNumber.from(currentLiquidity.toString())
+        : BigNumber.from(withdrawLiquidity.toString()),
+    [withdrawLiquidity, currentLiquidity, isFullWithdrawal],
+  );
+
+  const handleSubmit = useHandleSubmit(
+    withdraw,
+    isFullWithdrawal,
+    pool,
+    position,
+    onClose,
+  );
+  const isValidAmount = useMemo(
+    () => Number(withdrawAmount) <= Number(depositedAmountBase),
+    [withdrawAmount, depositedAmountBase],
+  );
+
+  const isSubmitDisabled = useMemo(
+    () =>
+      !depositedAmountBase ||
+      !withdrawAmount ||
+      Number(withdrawAmount) <= 0 ||
+      !isValidAmount,
+    [depositedAmountBase, withdrawAmount, isValidAmount],
+  );
+
+  useEffect(() => {
+    const getDepositedAmounts = async () => {
+      if (!croc || !poolTokens) {
+        setDepositedAmountBase(Decimal.ZERO);
+        return;
+      }
+      const pool = croc.pool(
+        poolTokens.tokenA.tokenAddr,
+        poolTokens.tokenB.tokenAddr,
+      );
+      const spotPrice = await pool.spotPrice();
+      const sqrtPrice = Math.sqrt(spotPrice);
+      setPoolPrice(sqrtPrice);
+      const baseAmount = decimalic(currentLiquidity).mul(sqrtPrice);
+      const quoteAmount = decimalic(currentLiquidity).div(sqrtPrice);
+      const convertedBaseAmount = baseAmount.div(Math.pow(10, 18));
+      const convertedQuoteAmount = quoteAmount.div(Math.pow(10, 18));
```

Comment:
Shouldn't this be token decimals instead of 18?

Reply:
@pietro-maximoff please update